server/api.ts

```diff
@@ -1,22 +1,36 @@
-import { Post, User } from './types';
+import { Post, User,Like,Comment } from './types';
 
 // Post APIs
 export interface ListPostsRequest {}
 export interface ListPostsResponse {
   posts: Post[];
 }
-
+export type DeletePostRequest = { postId: string };
+export type DeletePostResponse = {};
 export type CreatePostRequest = Pick<Post, 'title' | 'url' | 'userId'>;
 export interface CreatePostResponse {}
-
-export interface GetPostRequest {}
+export type GetPostRequest = {postId:string}
 export interface GetPostResponse {
   post: Post;
 }
 
 // Comment APIs
+export type CreateCommentRequest = Pick<Comment, 'userId' | 'postId' | 'comment'>;
+export interface CreateCommentResponse {}
+export type GetCommentsRequest = { postId: string | undefined };
```

Comment:
you don't need the ` | undefined` here, the type we created makes sure all these fields are optional, since the server can't really be sure the client sticks to using the types.